flash_attn/models/gpt.py

```diff
@@ -574,6 +574,7 @@ def __init__(self, config: GPT2Config, process_group=None, device=None, dtype=No
         super().__init__(config)
         self.process_group = process_group
         self.transformer = GPTModel(config, process_group=process_group, **factory_kwargs)
+        self.device = torch.device(device)
```

Comment:
What happens if we construct the model and then do `mode = model.to('cuda')`? Idk how hf keeps track of self.device?

Reply:
Good question. Looks like it uses this:

https://github.com/huggingface/transformers/blob/74081cb5fa52540bbdde620942bd3a657af85c8e/src/transformers/modeling_utils.py#L821

Maybe we can just re-use that mix-in.

I'm thinking instead of modifying `GPTLMHeadModel`, I should subclass it into a HF compat class that adds the required methods.

Reply:
That sounds better to me. I personally don't use HF trainer. I didn't realize HF trainer doesn't just take an nn.Module but requires the class to implement specific methods.